format/amf3/Tools.hx

```diff
@@ -87,13 +95,67 @@ class Tools {
 			case AString(_): string(a);
 			case ADate(_): date(a);
 			case AArray(_): array(a);
+			case AVector(_): vector(a);
 			case AObject(_,_): object(a);
 			case AXml(_): xml(a);
 			case ABytes(_): bytes(a);
 			case AMap(_): map(a);
 		}
 	}
 
+	public static function decodeWithAnonymousStruct(o:Value):Dynamic
+	{
+		switch(o)
+		{
+			case AObject(f,_):
+				var s:Dynamic = {};
+				for( k in f.keys() )
+				{
+					var so:Value = f[k];
+					var ss:Dynamic = decodeWithAnonymousStruct(so);
+					Reflect.setField(s, k, ss);
+				}
+				return s;
+			case AArray(oa):
+				var a:Array<Dynamic> = [];
+				// get both associative array and indexed/dense array values
+				for (af in Reflect.fields(oa))
+				{
+					Reflect.setField(a, af, decodeWithAnonymousStruct(Reflect.field(oa, af)));
+				}
```

Comment:
I don't think it should be #if, maybe you can have your own version of this function in your project, and keep the original version as-it?